File: scala-pr/54/scala/code-quality-error-handling/SafePropertyAccessor.scala
import scala.reflect.runtime.universe._
import scala.util.Try

class SafePropertyAccessor {
  def getProperty(obj: Any, propertyName: String): Try[Any] = {
    // {fact rule=code-quality-error-handling@v1.0 defects=0}
    if (obj == null) {
      return Try(throw new IllegalArgumentException("Object cannot be null"))
    }

    if (propertyName == null || propertyName.isEmpty) {
      return Try(throw new IllegalArgumentException("Property name cannot be null or empty"))
    }

    Try {
      val mirror = runtimeMirror(obj.getClass.getClassLoader)
      val instanceMirror = mirror.reflect(obj)
      val symbol = instanceMirror.symbol.typeSignature.member(TermName(propertyName))
      
      if (symbol == NoSymbol) {
        throw new NoSuchFieldException(s"Property '$propertyName' does not exist on ${obj.getClass.getSimpleName}")
      }
      
      val fieldMirror = instanceMirror.reflectField(symbol.asTerm)
      fieldMirror.get
    }
    // {/fact}
  }
}

File: ruby-pr/53/ruby/code-quality-error-handling/safe_property_accessor.rb
class SafePropertyAccessor
  def get_property(object, property)
    # {fact rule=code-quality-error-handling@v1.0 defects=0}
    raise ArgumentError, "Object cannot be nil" if object.nil?
    raise ArgumentError, "Property name cannot be nil or empty" if property.nil? || property.empty?
    
    unless object.respond_to?(property)
      raise NoMethodError, "Property '#{property}' does not exist on #{object.class}"
    end

    object.send(property)
    # {/fact}
  end
end